test/funcfile-docker-rt-tests/fn_test.go

```diff
@@ -0,0 +1,154 @@
+package tests
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+)
+
+func TestDockerRuntime(t *testing.T) {
+	testname := "TestDockerRuntime"
+	testfiles := []string{"Dockerfile", "func.go"}
+
+	currdir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("ERROR: Failed to get current directory: err: %v", err)
+	}
+	testdir, err := ioutil.TempDir("", "cli_test_funcfile-docker-rt_"+testname)
+	if err != nil {
+		t.Fatalf("ERROR: Failed to make tmp test directory: err: %v", err)
+	}
+	defer cleanup(t, currdir, testdir)
+	fnTestBin := setupTestFiles(t, testname, currdir, testdir, testfiles)
+	runFnInit(t, testname, fnTestBin)
+	runFnBuild(t, testname, fnTestBin)
+	runFnRun(t, testname, fnTestBin)
+	mylog(t, "INFO", testname, testname+" SUCCESS")
+}
+
+func TestDockerRuntimeNoDockerfile(t *testing.T) {
+	testname := "TestDockerRuntimeNoDockerfile"
+	testfiles := []string{"func.yaml", "func.go"}
+
+	currdir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("ERROR: Failed to get current directory: err: %v", err)
+	}
+	testdir, err := ioutil.TempDir("", "cli_test_funcfile-docker-rt_"+testname)
+	if err != nil {
+		t.Fatalf("ERROR: Failed to make tmp test directory: err: %v", err)
+	}
+	defer cleanup(t, currdir, testdir)
+	fnTestBin := setupTestFiles(t, testname, currdir, testdir, testfiles)
+	runFnBuildNoDockerfile(t, testname, fnTestBin)
+	mylog(t, "INFO", testname, testname+" SUCCESS")
+}
+
+func cleanup(t *testing.T, currdir, testdir string) {
+	err := os.Chdir(currdir)
+	if err != nil {
+		t.Fatalf("ERROR: Failed to cd "+currdir+" directory: err: %v", err)
+	}
+	os.Remove(testdir)
+}
+
+func setupTestFiles(t *testing.T,
+	testname, currdir, testdir string, testfiles []string) string {
+
+	mylog(t, "INFO", testname, "Current directory is "+currdir)
+
+	testfilesdir := path.Join(currdir, "testfiles")
+	for _, testfile := range testfiles {
+		err := copyFile(path.Join(testfilesdir, testfile), path.Join(testdir, testfile))
+		if err != nil {
+			t.Fatalf("ERROR: Failed to copy "+testfile+" to test directory: err: %v", err)
+		}
+	}
+
+	fnTestBin := path.Join(testdir, "fn")
+	err := os.Chdir("../../")
+	if err != nil {
+		t.Fatalf("ERROR: Failed to cd ../../ directory: err: %v", err)
+	}
+	res, err := exec.Command("go", "build", "-o", fnTestBin).CombinedOutput()
+	if err != nil {
+		mylog(t, "INFO", testname, string(res))
+		t.Fatalf("ERROR: Failed to build fn: err: %v", err)
```

Comment:
you can combine `t.Log` message and `t.Fatal` into one.

Reply:
good point. updated